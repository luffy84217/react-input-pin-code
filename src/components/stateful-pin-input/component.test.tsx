import { describe, expect, it, jest } from '@jest/globals';
import { fireEvent, render } from '@testing-library/react';

import { StatefulPinInput } from '.';
import React, { Dispatch, SetStateAction } from 'react';

describe('component: StatefulPinInput', () => {
  it('renders component', () => {
    const { asFragment } = render(
      <StatefulPinInput
        initialValue={['1', '2', '3', '4']} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should contain 4 inputs by default', () => {
    const { getAllByRole } = render(<StatefulPinInput />);
    const PinInputFields = getAllByRole('textbox');

    expect(PinInputFields).toHaveLength(4);
  });

  it('should update state on change', () => {
    const changePin = jest.fn();
    const { getAllByRole } = render(<StatefulPinInput onChange={changePin} />);
    const handleChange = jest.spyOn(React, 'useState') as jest.SpiedFunction<
      (initialState: string[]) => [string[], Dispatch<SetStateAction<string[]>>]
    >;
    handleChange.mockImplementation((initialState) => [initialState, changePin]);

    const PinInputFields = getAllByRole('textbox') as HTMLInputElement[];

    PinInputFields.forEach((input, i) => {
      fireEvent.change(input, { target: { value: i } });
    });

    expect(changePin).toBeTruthy();
  });
});
