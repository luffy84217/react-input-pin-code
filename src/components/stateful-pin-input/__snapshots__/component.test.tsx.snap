// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: StatefulPinInput renders component 1`] = `
<DocumentFragment>
  <div
    class="pinCodeContainer"
  >
    <input
      aria-label="Please enter pin code"
      autocomplete="off"
      class="inputWrapper isMedium hasCompleted showState"
      data-index="0"
      inputmode="numeric"
      placeholder="o"
      type="text"
      value="1"
    />
    <input
      aria-label="Please enter pin code"
      autocomplete="off"
      class="inputWrapper isMedium hasCompleted showState"
      data-index="1"
      inputmode="numeric"
      placeholder="o"
      type="text"
      value="2"
    />
    <input
      aria-label="Please enter pin code"
      autocomplete="off"
      class="inputWrapper isMedium hasCompleted showState"
      data-index="2"
      inputmode="numeric"
      placeholder="o"
      type="text"
      value="3"
    />
    <input
      aria-label="Please enter pin code"
      autocomplete="off"
      class="inputWrapper isMedium hasCompleted showState"
      data-index="3"
      inputmode="numeric"
      placeholder="o"
      type="text"
      value="4"
    />
  </div>
</DocumentFragment>
`;
